orchagent/aclorch.cpp

```diff
@@ -809,8 +809,22 @@ bool AclRuleL3::validateAddAction(string attr_name, string _attr_value)
         // handle PACKET_ACTION_REDIRECT in ACTION_PACKET_ACTION for backward compatibility
         else if (attr_value.find(PACKET_ACTION_REDIRECT) != string::npos)
         {
-            // resize attr_value to remove argument, _attr_value still has the argument
-            attr_value.resize(string(PACKET_ACTION_REDIRECT).length());
+            // check that we have a colon after redirect rule
+            size_t colon_pos = string(PACKET_ACTION_REDIRECT).length();
+
+            if (attr_value.c_str()[colon_pos] != ':')
```

Comment:
would `attr_value[colon_pos]` work here? I don't understand why do we need to extract c_str from the string?

Reply:
if size of attr_value is colon_pos (e.g. attr_value == PACKET_ACTION_REDIRECT), then accessing attr_value[colon_pos] is accessing out of bounds, so undefined behaviour. std::string is not required to put '\0' at the end of the string while c_str() method is. So in case size of attr_value is colon_pos then attr_value.c_str()[colon_pos] will return null byte.